packages/swapper/src/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts

```diff
@@ -28,59 +31,84 @@ export const getNetworkFeeCryptoBaseUnit = async ({
 
   const { average } = await adapter.getGasFeeData()
 
-  const l1GasLimit = await (async () => {
-    if (!L1_FEE_CHAIN_IDS.includes(chainId as KnownChainIds)) return
-
+  try {
+    if (!from) throw new Error('Cannot estimate fees without from')
     const { transactionRequest } = await getStepTransaction(lifiStep)
-    const { data, gasLimit } = transactionRequest ?? {}
-
-    if (!data || !gasLimit) {
-      throw new Error('getStepTransaction failed')
-    }
-
-    const publicClient = viemClientByChainId[chainId as EvmChainId]
-
-    const abi = [
-      {
-        inputs: [{ internalType: 'bytes', name: '_data', type: 'bytes' }],
-        name: 'getL1GasUsed',
-        outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
-        stateMutability: 'view',
-        type: 'function',
-      },
-    ]
-
-    const contract = getContract({
-      address: L1_GAS_ORACLE_ADDRESS,
-      abi,
-      client: {
-        public: publicClient,
-      },
+    if (!transactionRequest) throw new Error('transactionRequest is undefined')
+
+    const { data, to, value } = transactionRequest
+
+    if (data === undefined) throw new Error('transactionRequest: data is required')
+    if (to === undefined) throw new Error('transactionRequest: to is required')
+    if (value === undefined) throw new Error('transactionRequest: value is required')
+
+    // Attempt own fees estimation if possible (approval granted, wallet connected with from addy to use for simulation)
+    const feeData = await evm.getFees({
+      adapter,
+      data,
+      to,
+      value: bn(value.toString()).toString(),
```

Comment:
[suggestion]: Value is already a string

```suggestion
      value,
```

Reply:
Unless the type system is lying (not unlikely)?

Reply:
Monkey saw monkey did 🐒 confirmed we're still gucci
c4a0ef820d